pkg/apiserver/handlers/featuregates/handler.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package featuregates
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"gopkg.in/yaml.v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/controller/querier"
+	"antrea.io/antrea/pkg/features"
+	"antrea.io/antrea/pkg/util/env"
+)
+
+var controllerOnlyGates = []string{
```

Comment:
It could use sets.String to simplify `isControlGates` (the function will not be needed)

Reply:
sure, will refine this.

Reply:
done